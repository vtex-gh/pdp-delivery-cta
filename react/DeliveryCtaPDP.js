import DeliveryCtaPDP from './components/DeliveryCtaPDP/index'

export default DeliveryCtaPDP
