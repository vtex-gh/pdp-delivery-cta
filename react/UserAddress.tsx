import React, { useEffect } from 'react'
import { useQuery } from 'react-apollo'
import { useCssHandles } from 'vtex.css-handles'
import type { CssHandlesTypes } from 'vtex.css-handles'
import ADDRESS_QUERY from 'vtex.store-resources/QueryAddress'

import Container from './Container'
import AddressInfo, {
  CSS_HANDLES as AddressInfoCssHandles,
} from './components/UserAddress/AddressInfo'
import { UserAddressCssHandlesProvider } from './components/UserAddress/UserAddressCssHandles'

export const USER_ADDRESS_CSS_HANDLES = [
  'userAddressContainer',
  ...AddressInfoCssHandles,
] as const

type Props = {
  /** Define if address should be displayed in a single line or in blocks */
  variation: 'inline' | 'bar'
  /** Define if street should be displayed */
  showStreet?: boolean
  /** Define if city and state should be displayed */
  showCityAndState?: boolean
  /** Define if postal code should be displayed */
  showPostalCode?: boolean
  /** Define if prefix should be displayed */
  showPrefix?: boolean
  /** Define if component should render if address is empty */
  showIfEmpty?: boolean
  /** Used to override default CSS handles */
  classes?: CssHandlesTypes.CustomClasses<typeof USER_ADDRESS_CSS_HANDLES>
}

function UserAddress({
  variation,
  showStreet = true,
  showCityAndState = false,
  showPostalCode = false,
  showPrefix = true,
  showIfEmpty = false,
  classes,
}: Props) {
  const { handles, withModifiers } = useCssHandles(USER_ADDRESS_CSS_HANDLES, {
    classes,
  })

  // todo: type the data returned
  const { data: orderFormData, refetch } = useQuery(ADDRESS_QUERY, {
    ssr: false,
  })

  useEffect(() => {
    const handleLocationUpdated = () => refetch()

    window.addEventListener('locationUpdated', handleLocationUpdated)

    return () => {
      window.removeEventListener('locationUpdated', handleLocationUpdated)
    }
  }, [refetch])

  const orderForm = orderFormData?.orderForm

  if (!orderForm || (!orderForm?.shippingData?.address && !showIfEmpty)) {
    return null
  }

  const isInline = variation === 'inline'
  const isInverted = !isInline

  const AddressInfoWithHandles = () => {
    return (
      <UserAddressCssHandlesProvider
        handles={handles}
        withModifiers={withModifiers}
      >
        <AddressInfo
          inverted={isInverted}
          inline={isInline}
          orderForm={orderForm}
          showStreet={showStreet}
          showCityAndState={showCityAndState}
          showPostalCode={showPostalCode}
          showPrefix={showPrefix}
          showIfEmpty={showIfEmpty}
        />
      </UserAddressCssHandlesProvider>
    )
  }

  if (isInline) {
    return (
      <div
        className={`${handles.userAddressContainer} ph5`}
        style={{
          maxWidth: '30rem',
        }}
      >
        <AddressInfoWithHandles />
      </div>
    )
  }

  return (
    <div
      className={`${handles.userAddressContainer} bg-base--inverted c-on-base--inverted flex ph5 pointer pv3 ml3 mr3`}
    >
      <Container className="flex justify-center w-100 left-0">
        <div className="w-100 mw9 flex">
          <AddressInfoWithHandles />
        </div>
      </Container>
    </div>
  )
}

export default UserAddress
