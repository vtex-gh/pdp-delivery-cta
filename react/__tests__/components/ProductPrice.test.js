import React from 'react'
import { render } from '@vtex/test-tools/react'

import ProductPrice from '../../ProductPrice'

describe('<ProductPrice />', () => {
  const defaultProps = {
    loaderClass: '',
    showListPrice: false,
    sellingPrice: 40,
  }

  const renderComponent = customProps => {
    const props = {
      ...defaultProps,
      ...customProps,
    }

    return render(<ProductPrice {...props} />)
  }

  it('should be mount', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toBeDefined()
  })

  it('should match snapshot', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot Loader', () => {
    const { asFragment } = renderComponent({ showListPrice: true })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should show selling price and list price (with labels)', () => {
    const customProps = {
      listPrice: 50,
      showLabels: true,
      showListPrice: true,
      labelSellingPrice: 'To',
      labelListPrice: 'From',
    }

    const { getByText } = renderComponent(customProps)

    expect(getByText('From')).toBeDefined()
    expect(getByText('To')).toBeDefined()
    expect(getByText('$40.00')).toBeDefined()
    expect(getByText('$50.00')).toBeDefined()
  })

  it('should show selling price and list price but with NO labels', () => {
    const customProps = {
      listPrice: 50,
      showLabels: false,
      showListPrice: true,
      labelSellingPrice: 'To',
      labelListPrice: 'From',
    }

    const { getByText, queryByText } = renderComponent(customProps)

    expect(queryByText('From')).toBeNull()
    expect(queryByText('To')).toBeNull()
    expect(getByText('$40.00')).toBeDefined()
    expect(getByText('$50.00')).toBeDefined()
  })

  // eslint-disable-next-line jest/no-identical-title
  it('should show selling price and list price but with NO labels', () => {
    const customProps = {
      listPrice: 50,
      showLabels: false,
      showListPrice: true,
    }

    const { getByText, queryByText } = renderComponent(customProps)

    expect(queryByText('From')).toBeNull()
    expect(queryByText('To')).toBeNull()
    expect(getByText('$40.00')).toBeDefined()
    expect(getByText('$50.00')).toBeDefined()
  })

  it('should show selling price and list price but with custom labels', () => {
    const customProps = {
      listPrice: 50,
      showLabels: true,
      showListPrice: true,
      labelSellingPrice: 'Now',
      labelListPrice: 'Was',
    }

    const { getByText } = renderComponent(customProps)

    expect(getByText('Was')).toBeDefined()
    expect(getByText('Now')).toBeDefined()
    expect(getByText('$40.00')).toBeDefined()
    expect(getByText('$50.00')).toBeDefined()
  })

  it('should show range selling price (not list price)', () => {
    const customProps = {
      listPrice: 50,
      showLabels: true,
      showListPrice: false,
      labelSellingPrice: 'Now',
      labelListPrice: 'Was',
      sellingPriceList: [30, 50, 20, 55],
      showSellingPriceRange: true,
    }

    const { getByText, queryByText } = renderComponent(customProps)

    // dont show labels
    expect(queryByText('Now')).toBeNull()
    expect(queryByText('Was')).toBeNull()
    // dont show selling price
    expect(queryByText('$40.00')).toBeNull()

    expect(getByText('$20.00 - $55.00')).toBeDefined()
  })

  it('show list price range and selling price range (no labels)', () => {
    const customProps = {
      listPrice: 50,
      showLabels: false,
      showListPrice: true,
      showListPriceRange: true,
      labelSellingPrice: 'Now',
      labelListPrice: 'Was',
      sellingPriceList: [30, 50, 20, 55],
      listPriceList: [100, 200, 300, 400],
      showSellingPriceRange: true,
    }

    const { getByText, queryByText } = renderComponent(customProps)

    // dont show labels
    expect(queryByText('Now')).toBeNull()
    expect(queryByText('Was')).toBeNull()
    // dont show selling price
    expect(queryByText('$40.00')).toBeNull()
    expect(queryByText('$50.00')).toBeNull()

    expect(getByText('$20.00 - $55.00')).toBeDefined()
    expect(getByText('$100.00 - $400.00')).toBeDefined()
  })

  it('show list price range and selling price range (with labels)', () => {
    const customProps = {
      listPrice: 50,
      showLabels: true,
      showListPrice: true,
      showListPriceRange: true,
      labelSellingPrice: 'Now',
      labelListPrice: 'Was',
      sellingPriceList: [30, 50, 20, 55],
      listPriceList: [100, 200, 300, 400],
      showSellingPriceRange: true,
    }

    const { getByText, queryByText } = renderComponent(customProps)

    // show labels
    expect(getByText('Now')).toBeDefined()
    expect(getByText('Was')).toBeDefined()
    // dont show selling price
    expect(queryByText('$40.00')).toBeNull()
    expect(queryByText('$50.00')).toBeNull()

    expect(getByText('$20.00 - $55.00')).toBeDefined()
    expect(getByText('$100.00 - $400.00')).toBeDefined()
  })
})
