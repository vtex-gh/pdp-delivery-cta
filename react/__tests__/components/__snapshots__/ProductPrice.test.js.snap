// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPrice /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="priceContainer price_className"
  >
    <div
      class="sellingPrice sellingPriceContainer price_sellingPriceContainer"
    >
      <span
        class="sellingPrice sellingPriceValue price_sellingPrice"
      >
        $40.00
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ProductPrice /> should match snapshot Loader 1`] = `
<DocumentFragment>
  <div
    class="priceContainer priceLoaderContainer  price_loader"
  >
    <svg
      class="content-loader-mock"
    >
      <rect
        height="0.75em"
        width="50%"
        x="25%"
      />
      <rect />
      <rect
        height="1em"
        width="70%"
        x="15%"
        y="1.25em"
      />
      <rect
        height="0.75em"
        width="80%"
        x="10%"
        y="2.75em"
      />
      <rect />
    </svg>
  </div>
</DocumentFragment>
`;
